llvm/lib/CodeGen/SelectionDAG/FunctionLoweringInfo.cpp

```diff
@@ -69,6 +69,20 @@ static ISD::NodeType getPreferredExtendForValue(const Instruction *I) {
       NumOfSigned += CI->isSigned();
       NumOfUnsigned += CI->isUnsigned();
     }
+    if (const auto *CallI = dyn_cast<CallInst>(U)) {
```

Comment:
If you switch the loop to be over `Use &` rather than `User *`, you can avoid iterating over the call arguments.

Reply:
Also, this should be using CallBase, not CallInst, to also cover invokes.

Reply:
Sorry, I don't understand how to avoid iterating over the call arguments. Do you mean that I need to change the `User *` to `User &` on Line 67?

Reply:
Yes, and change I->users to I->uses

Reply:
I have changed `CallInst` to `CallBase`, and  switch the loop to be over `Use &`.